scrapy/http/request/form.py

```diff
@@ -33,7 +33,11 @@ def __init__(self, *args, **kwargs):
                 self.headers.setdefault(b'Content-Type', b'application/x-www-form-urlencoded')
                 self._set_body(querystr)
             else:
-                self._set_url(self.url + ('&' if '?' in self.url else '?') + querystr)
+                query = dict(parse_qsl(urlparse(self.url).query))
+                if query:
+                    query.update(dict(items))
```

Comment:
Sometimes we've to send multiple values for a single key, `dict(items)` will make it impossible and will result in unexpected URL.

`
formdata=[
('a','1'),
('a','2'),
('b','3'),
]
`